x-pack/legacy/plugins/infra/common/http_api/node_details_api.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as rt from 'io-ts';
+
+const NodeDetailsDataPointRT = rt.intersection([
+  rt.type({
+    timestamp: rt.number,
+  }),
+  rt.partial({
+    value: rt.number,
+  }),
+]);
+
+const NodeDetailsDataSeries = rt.type({
+  id: rt.string,
+  label: rt.string,
+  data: rt.array(NodeDetailsDataPointRT),
+});
+
+const NodeDetailsMetricsRT = rt.keyof({
```

Comment:
Couldn't we use `InventoryMetricRT` and `InventoryMetric`  respectively from `common/inventory_models/types`?